File: CaseStudies/SolarHeatingWaterOnly/CCode/Testing/src/compareMatlabTest.c
#include "unity.h"
#include "unity_fixture.h"
#include "PCM_Error.h"
#include "parameters.h"
#include "load_params.h"

TEST_GROUP(CompareMatlab);

TEST_SETUP(CompareMatlab){
}

TEST_TEAR_DOWN(CompareMatlab){
}

TEST(CompareMatlab, testCompareMatlab1){
    struct parameters params;
    params = load_params("Testing/src/inputFiles/C01.in");
    double errTw, errTp, errEw, errEp;
    double delta = 0.00001;
    errTw = PCM_ErrorM("Testing/src/compareMatlab/M01.out", "C01.out", "TWat", params);
    errTp = PCM_ErrorM("Testing/src/compareMatlab/M01.out", "C01.out", "TPCM", params);
    errEw = PCM_ErrorM("Testing/src/compareMatlab/M01.out", "C01.out", "EWat", params);
    errEp = PCM_ErrorM("Testing/src/compareMatlab/M01.out", "C01.out", "EPCM", params);
    TEST_ASSERT_FLOAT_WITHIN_MESSAGE(delta, 0, errTw, "Water temperature");
    TEST_ASSERT_FLOAT_WITHIN_MESSAGE(delta, 0, errTp, "PCM temperature");
    TEST_ASSERT_FLOAT_WITHIN_MESSAGE(delta, 0, errEw, "Water energy");
    TEST_ASSERT_FLOAT_WITHIN_MESSAGE(delta, 0, errEp, "PCM energy");
}

TEST(CompareMatlab, testCompareMatlab2){
    struct parameters params;
    params = load_params("Testing/src/inputFiles/C02.in");
    double errTw, errTp, errEw, errEp;
    double delta = 0.00001;
    errTw = PCM_ErrorM("Testing/src/compareMatlab/M02.out", "C02.out", "TWat", params);
    errTp = PCM_ErrorM("Testing/src/compareMatlab/M02.out", "C02.out", "TPCM", params);
    errEw = PCM_ErrorM("Testing/src/compareMatlab/M02.out", "C02.out", "EWat", params);
    errEp = PCM_ErrorM("Testing/src/compareMatlab/M02.out", "C02.out", "EPCM", params);
    TEST_ASSERT_FLOAT_WITHIN_MESSAGE(delta, 0, errTw, "Water temperature");
    TEST_ASSERT_FLOAT_WITHIN_MESSAGE(delta, 0, errTp, "PCM temperature");
    TEST_ASSERT_FLOAT_WITHIN_MESSAGE(delta, 0, errEw, "Water energy");
    TEST_ASSERT_FLOAT_WITHIN_MESSAGE(delta, 0, errEp, "PCM energy");
}

TEST(CompareMatlab, testCompareMatlab3){
    struct parameters params;
    params = load_params("Testing/src/inputFiles/C03.in");
    double errTw, errTp, errEw, errEp;
    double delta = 0.00001;
    errTw = PCM_ErrorM("Testing/src/compareMatlab/M03.out", "C03.out", "TWat", params);
    errTp = PCM_ErrorM("Testing/src/compareMatlab/M03.out", "C03.out", "TPCM", params);
    errEw = PCM_ErrorM("Testing/src/compareMatlab/M03.out", "C03.out", "EWat", params);
    errEp = PCM_ErrorM("Testing/src/compareMatlab/M03.out", "C03.out", "EPCM", params);
    TEST_ASSERT_FLOAT_WITHIN_MESSAGE(delta, 0, errTw, "Water temperature");
    TEST_ASSERT_FLOAT_WITHIN_MESSAGE(delta, 0, errTp, "PCM temperature");
    TEST_ASSERT_FLOAT_WITHIN_MESSAGE(delta, 0, errEw, "Water energy");
    TEST_ASSERT_FLOAT_WITHIN_MESSAGE(delta, 0, errEp, "PCM energy");
}

TEST(CompareMatlab, testCompareMatlab4){
    struct parameters params;
    params = load_params("Testing/src/inputFiles/C04.in");
    double errTw, errTp, errEw, errEp;
    double delta = 0.00001;
    errTw = PCM_ErrorM("Testing/src/compareMatlab/M04.out", "C04.out", "TWat", params);
    errTp = PCM_ErrorM("Testing/src/compareMatlab/M04.out", "C04.out", "TPCM", params);
    errEw = PCM_ErrorM("Testing/src/compareMatlab/M04.out", "C04.out", "EWat", params);
    errEp = PCM_ErrorM("Testing/src/compareMatlab/M04.out", "C04.out", "EPCM", params);
    TEST_ASSERT_FLOAT_WITHIN_MESSAGE(delta, 0, errTw, "Water temperature");
    TEST_ASSERT_FLOAT_WITHIN_MESSAGE(delta, 0, errTp, "PCM temperature");
    TEST_ASSERT_FLOAT_WITHIN_MESSAGE(delta, 0, errEw, "Water energy");
    TEST_ASSERT_FLOAT_WITHIN_MESSAGE(delta, 0, errEp, "PCM energy");
}

File: CaseStudies/SolarHeatingWaterOnly/Python/Testing/compareMatlabTest.py
import sys
sys.path.insert(0, '.')

import unittest
import PCM_Error


class TestCompareMatlab(unittest.TestCase):

    def setUp(self):
        self.delta = 0.000005

    def test_CM1(self):
        errTw = PCM_Error.PCM_ErrorM('Testing/compareMatlab/M01.out', 'P01.out', 'TWat')
        errTp = PCM_Error.PCM_ErrorM('Testing/compareMatlab/M01.out', 'P01.out', 'TPCM')
        errEw = PCM_Error.PCM_ErrorM('Testing/compareMatlab/M01.out', 'P01.out', 'EWat')
        errEp = PCM_Error.PCM_ErrorM('Testing/compareMatlab/M01.out', 'P01.out', 'EPCM')
        self.assertAlmostEqual(errTw, 0, places=None, msg='Water temperature', delta=self.delta)
        self.assertAlmostEqual(errTp, 0, places=None, msg='PCM temperature', delta=self.delta)
        self.assertAlmostEqual(errEw, 0, places=None, msg='Water energy', delta=self.delta)
        self.assertAlmostEqual(errEp, 0, places=None, msg='PCM energy', delta=self.delta)

    def test_CM2(self):
        errTw = PCM_Error.PCM_ErrorM('Testing/compareMatlab/M02.out', 'P02.out', 'TWat')
        errTp = PCM_Error.PCM_ErrorM('Testing/compareMatlab/M02.out', 'P02.out', 'TPCM')
        errEw = PCM_Error.PCM_ErrorM('Testing/compareMatlab/M02.out', 'P02.out', 'EWat')
        errEp = PCM_Error.PCM_ErrorM('Testing/compareMatlab/M02.out', 'P02.out', 'EPCM')
        self.assertAlmostEqual(errTw, 0, places=None, msg='Water temperature', delta=self.delta)
        self.assertAlmostEqual(errTp, 0, places=None, msg='PCM temperature', delta=self.delta)
        self.assertAlmostEqual(errEw, 0, places=None, msg='Water energy', delta=self.delta)
        self.assertAlmostEqual(errEp, 0, places=None, msg='PCM energy', delta=self.delta)

    def test_CM3(self):
        errTw = PCM_Error.PCM_ErrorM('Testing/compareMatlab/M03.out', 'P03.out', 'TWat')
        errTp = PCM_Error.PCM_ErrorM('Testing/compareMatlab/M03.out', 'P03.out', 'TPCM')
        errEw = PCM_Error.PCM_ErrorM('Testing/compareMatlab/M03.out', 'P03.out', 'EWat')
        errEp = PCM_Error.PCM_ErrorM('Testing/compareMatlab/M03.out', 'P03.out', 'EPCM')
        self.assertAlmostEqual(errTw, 0, places=None, msg='Water temperature', delta=self.delta)
        self.assertAlmostEqual(errTp, 0, places=None, msg='PCM temperature', delta=self.delta)
        self.assertAlmostEqual(errEw, 0, places=None, msg='Water energy', delta=self.delta)
        self.assertAlmostEqual(errEp, 0, places=None, msg='PCM energy', delta=self.delta)

    def test_CM4(self):
        errTw = PCM_Error.PCM_ErrorM('Testing/compareMatlab/M04.out', 'P04.out', 'TWat')
        errTp = PCM_Error.PCM_ErrorM('Testing/compareMatlab/M04.out', 'P04.out', 'TPCM')
        errEw = PCM_Error.PCM_ErrorM('Testing/compareMatlab/M04.out', 'P04.out', 'EWat')
        errEp = PCM_Error.PCM_ErrorM('Testing/compareMatlab/M04.out', 'P04.out', 'EPCM')
        self.assertAlmostEqual(errTw, 0, places=None, msg='Water temperature', delta=self.delta)
        self.assertAlmostEqual(errTp, 0, places=None, msg='PCM temperature', delta=self.delta)
        self.assertAlmostEqual(errEw, 0, places=None, msg='Water energy', delta=self.delta)
        self.assertAlmostEqual(errEp, 0, places=None, msg='PCM energy', delta=self.delta)


class CompareMatlabSuite:

    def suite(self):
        suite = unittest.TestLoader().loadTestsFromTestCase(TestCompareMatlab)
        return suite